language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -180,7 +180,7 @@ genDefDataType curPkgId conName curModName tpls def =
                                 (keyTypeTs, "() => " <> keySer <> ".decoder()")
                         dict =
                             ["export const " <> conName <> ": daml.Template<" <> conName <> ", " <> keyTypeTs <> "> & {"] ++
-                            ["  " <> x <> ": daml.Choice<" <> conName <> ", " <> t <> ", " <> rtyp <> " >;" | (x, t, rtyp, _) <- chcs] ++
```

Comment:
Looks like somebody burnt his fingers on the `<X<Y>>` vs `<X<Y> >` quirk in C++ in the past. This is not an issue in TypeScript. :)

Reply:
i know. it's like drilled into my brain. it's not even an issue in "modern" C++ anymore.